plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/file/FileHiveMetastore.java

```diff
@@ -188,6 +192,49 @@ public FileHiveMetastore(NodeVersion nodeVersion, HdfsEnvironment hdfsEnvironmen
         }
     }
 
+    @Override
+    public synchronized MetastoreLock acquireTableExclusiveLock(
+            HiveIdentity identity,
+            AcidTransactionOwner transactionOwner,
+            String queryId,
+            String dbName,
+            String tableName)
+    {
+        SchemaTableName schemaTableName = new SchemaTableName(dbName, tableName);
+
+        ReentrantLock lock = tableLocks.computeIfAbsent(schemaTableName, ignored -> new ReentrantLock());
+        lock.lock();
+
+        tableLocks.put(schemaTableName, lock);
+        return new FileMetastoreLock(schemaTableName);
+    }
+
+    private class FileMetastoreLock
+            implements MetastoreLock
+    {
+        private final SchemaTableName tableName;
+        boolean released;
+
+        public FileMetastoreLock(SchemaTableName tableName)
+        {
+            this.tableName = requireNonNull(tableName, "tableName is null");
+        }
+
+        public void release()
+        {
+            if (released) {
+                throw new IllegalStateException("Table lock has already been released");
+            }
+            released = true;
+            ReentrantLock lock = tableLocks.get(tableName);
+            // Currently, metastore locks are always acquired and released in the same thread.
+            lock.unlock();
+            if (!lock.isLocked()) {
```

Comment:
`isLocked` is available on ReentrantLock, but not the Lock interface, so I did not apply that comment https://github.com/trinodb/trino/pull/12026#discussion_r859740124

Reply:
Maybe I am missing something but between checking if lock is not locked and removing it lock can locked again by another thread.